cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/transaction/coprocessor/hbase11/DefaultTransactionProcessor.java

```diff
@@ -83,55 +87,85 @@ public void stop(CoprocessorEnvironment e) throws IOException {
   }
 
   @Override
+  public void postFlush(ObserverContext<RegionCoprocessorEnvironment> e) throws IOException {
+    reloadPruneState(e.getEnvironment());
+    super.postFlush(e);
+  }
+
+  @Override
+  public InternalScanner preCompactScannerOpen(ObserverContext<RegionCoprocessorEnvironment> c, Store store,
+                                               List<? extends KeyValueScanner> scanners, ScanType scanType,
+                                               long earliestPutTs, InternalScanner s,
+                                               CompactionRequest request) throws IOException {
+    reloadPruneState(c.getEnvironment());
+    return super.preCompactScannerOpen(c, store, scanners, scanType, earliestPutTs, s, request);
+  }
+
+  @Override
   protected void ensureValidTxLifetime(RegionCoprocessorEnvironment env, OperationWithAttributes op,
                                        @Nullable Transaction tx) throws IOException {
     if (tx == null) {
       return;
     }
 
-    long maxLifetimeMillis;
-    if (txMaxLifetimeMillis == null) {
-      Configuration conf = getConfiguration(env);
-      if (conf != null) {
-        this.txMaxLifetimeMillis = TimeUnit.SECONDS.toMillis(conf.getInt(TxConstants.Manager.CFG_TX_MAX_LIFETIME,
-                                                                         TxConstants.Manager.DEFAULT_TX_MAX_LIFETIME));
-        maxLifetimeMillis = this.txMaxLifetimeMillis;
-      } else {
+    long currMaxLifetimeinMillis;
+    Long maxLifetimeFromConf = cConfCache.getTxMaxLifetimeMillis();
+    if (maxLifetimeFromConf == null) {
+      if (txMaxLifetimeMillis == null) {
         // Get maxLifetimeMillis from transaction attributes
         byte[] maxLifetimeBytes = op.getAttribute(HBaseTable.TX_MAX_LIFETIME_MILLIS_KEY);
         if (maxLifetimeBytes != null) {
-          maxLifetimeMillis = Bytes.toLong(maxLifetimeBytes);
+          currMaxLifetimeinMillis = Bytes.toLong(maxLifetimeBytes);
         } else {
+          // If maxLifetimeMillis could not be found anywhere, just use the default value and log a warning.
           LOG.warn("txMaxLifetimeMillis is not available in client's operation attributes. " +
```

Comment:
This can lead to a lot of warning messages in logs. Can we limit the rate using the log limitter class?